pkg/detectors/hashicorpvaultauth/hashicorpvaultauth.go

```diff
@@ -0,0 +1,192 @@
+package hashicorpvaultauth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	roleIdPat = regexp.MustCompile(detectors.PrefixRegex([]string{"role_id", "role-id", "roleid"}) + `\b([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\b`)
+
+	secretIdPat = regexp.MustCompile(detectors.PrefixRegex([]string{"secret_id", "secret-id", "secretid"}) + `\b([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})\b`)
+
+	// Vault URL pattern - HashiCorp Cloud or any HTTPS/HTTP Vault endpoint
+	vaultUrlPat = regexp.MustCompile(`https?:\/\/[^\s\/]*\.hashicorp\.cloud(?::\d+)?(?:\/[^\s]*)?`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+func (s Scanner) Keywords() []string {
+	return []string{"role_id", "secret_id", "vault", "approle", "hashicorp"}
+}
+
+// FromData will find and optionally verify HashiCorp Vault AppRole secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	roleIds := extractRoleIds(dataStr)
+	secretIds := extractSecretIds(dataStr)
+	vaultUrls := extractVaultUrls(dataStr)
+
+	// If no names or secrets found, return empty results
+	if len(roleIds) == 0 || len(secretIds) == 0 || len(vaultUrls) == 0 {
+		return results, nil
+	}
+
+	// create combination results that can be verified
+	for _, roleId := range roleIds {
+		for _, secretId := range secretIds {
+			for _, vaultUrl := range vaultUrls {
+				s1 := detectors.Result{
+					DetectorType: detectorspb.DetectorType_HashiCorpVaultAuth,
+					Raw:          []byte(secretId),
+					RawV2:        []byte(fmt.Sprintf("role_id:%s,secret_id:%s,vault_url:%s", roleId, secretId, vaultUrl)),
```

Comment:
```suggestion
					RawV2:        []byte(fmt.Sprintf("%s:%s", roleId, secretId)),
```

We can add vault url in extra data.

Reply:
Alll these are necessary in order to test the secret_id and role_id, should we not keep all in the RawV2?